### DISCUSSION TITLE
Number of cross-sectional grains

### DISCUSSION THREAD

**Discussion by sajutabraham** [neperfepx/neper#355] [Q&A] [answered]
@rquey How can we obtain the number of grains in a cross-section at a particular X-Y plane in a rectangular 3D geometry? Also, how we can calculate the average number of cross-sectional grains in that geometry? Thanks in advance!

**Comment by rquey** [comment, marked as answer]
You can slice the tessellation and count the number of grains:

```
neper -T -n 1000 -transform "slice(0.5,0,0,1)" -stattess cellnb -o d
more d.sttess
```

To get an average value, you can script:

```
neper -T -n 1000 -o 3d

rm -f vals
for z in `seq 0.01 0.01 0.99`
do
  neper -T -loadtess 3d.tess -transform "slice($z,0,0,1)" -stattess cellnb -o 2d
  more 2d.sttess >> vals
done

awk '{sum+=$1} END {print "average = "sum/NR}' vals
```

This is just an example. You will generally need more cells, etc. to get relevant statistics.

Needs version 4.2.1-22 (small fix).

**Reply by sajutabraham** [reply]
@rquey There are some issues while using "-reg 1". I have used "neper -T -n 1000 -reg 1 -o 3d". Also tried with "neper -T -n "from_morpho" -reg 1 -rsel 0.6 -domain "cube(0.5,0.5,0.5)" -morpho "diameq:lognormal(0.04,0.01),1-sphericity:lognormal(0.145,0.03)" -o 3d". In both cases, the 2d.sttess file was not created, and hence no results! If "-reg 0" is used, it runs. Can you please make a comment on this issue? Neper version used is 4.2.1-26.
